### PR TITLE
Allow to retrieve total entries count

### PR DESCRIPTION
**Pull request by jkulak**
I added the method to retrieve total entries count and also fixed the public getTotalEntries Pagination method.

### CODE REVIEW

File: src/Api/Pagination.php [hunk 1]
```diff
@@ -148,7 +148,7 @@ public function setTotalPages( $totalPages ): void {
 	 * @return null|int
 	 */
 	public function getTotalEntries(): ?int {
-		return $this->totalPages;
+		return $this->totalPages * $this->perPage;
```
Comment (ocean90) [inline src/Api/Pagination.php:151]:
```suggestion
		return $this->totalEntries;
```

File: src/Api/AbstractApi.php [hunk 2]
```diff
@@ -64,6 +64,15 @@ public function getPreviousPage(): ?int {
 		return $this->pagination->getPreviousPage();
 	}
 
+	/**
+	 * Retrieves total entries (estimated since last page does not have to be full).
+	 *
+	 * @return null|int
+	 */
+	public function getTotalEntries(): ?int {
+		return $this->pagination->getPreviousPage();
```
Comment (ocean90) [inline src/Api/AbstractApi.php:73]:
```suggestion
		return $this->pagination->getTotalEntries();
```

Comment (jkulak) [reply src/Api/AbstractApi.php:73]: Yes.

File: src/Api/Pagination.php (hunk 1 above)
Comment (jkulak) [reply src/Api/Pagination.php:151]: Yes.

**Comment by ocean90** [comment]
@jkulak Looks like the commit with the changes is missing?

**Comment by jkulak** [comment]
Done.

**Comment by jkulak** [comment]
I fixed it, squashed it and force pushed it :)
✅ done.

File: src/Api/AbstractApi.php [hunk 3]
```diff
@@ -64,6 +64,15 @@ public function getPreviousPage(): ?int {
 		return $this->pagination->getPreviousPage();
 	}
 
+	/**
+	 * Retrieves total entries (estimated since last page does not have to be full).
```
Comment (ocean90) [inline src/Api/AbstractApi.php:68]:
```suggestion
	 * Retrieves total entries number.
```

**Comment by jkulak** [comment]
Anyway. I can't get the totalEntries value with `allWithAutoPagingIterator()`.

How would I retrieve that?

```php
// Grab all time entries with auto pagination
$items = $client->timeEntries()->allWithAutoPagingIterator();
// Here? I guess not, since there were no requests yet

foreach ($items as $index => $item) {
    // Or here?
    // $totalEntries = $client->timeEntries()->getTotalEntries(); // this returns null
    }
```

**Comment by ocean90** [comment]
The auto pagination iterator is designed to not worry about total entries/pages. If you need the number of total entries it's probably better to use `all()` with custom pagination. Something like this should work:

```php
$client->timeEntries()->setPerPage( 10 );
$first_ten_items = $client->timeEntries()->all();
$total_entries = $client->timeEntries()->getTotalEntries();
$client->timeEntries()->setPage( $client->timeEntries()->getNextPage() );
$next_ten_items = $client->timeEntries()->all();
```

**Comment by jkulak** [comment]
Thanks for that. I find `allWithAutoPagingIterator()` the biggest power of this wrapper. So I guess I will just add an extra request, to get totalEntries, since I "need" it for my progressBar.

```php
$client->timeEntries()->setPerPage( 1 );
$first_ten_items = $client->timeEntries()->all();
$total_entries = $client->timeEntries()->getTotalEntries();
```

and then

```php
$items = $client->timeEntries()->allWithAutoPagingIterator();
```

**Pull request merged**